tests/e2e/testRunner.ts

```diff
@@ -137,10 +144,33 @@ const runTests = async (): Promise<void> => {
         Logger.log('Launching', appPackage);
         await launchApp('android', appPackage, config.ACTIVITY_PATH, launchArgs);
 
+        start(appPackage);
         await withFailTimeout(
             new Promise<void>((resolve) => {
-                server.addTestDoneListener(() => {
+                const subscription = server.addTestDoneListener(() => {
                     Logger.success(iterationText);
+
+                    const metrics = stop();
+                    const test = server.getTestConfig();
+
+                    if (server.isReadyToAcceptTestResults) {
+                        attachTestResult({
+                            name: `${test.name} (CPU)`,
+                            branch,
+                            duration: metrics.cpu,
+                        });
+                        attachTestResult({
+                            name: `${test.name} (FPS)`,
+                            branch,
+                            duration: metrics.fps,
+                        });
+                        attachTestResult({
+                            name: `${test.name} (RAM)`,
+                            branch,
+                            duration: metrics.ram,
+                        });
+                    }
+                    subscription();
```

Comment:
here we remove the listener when we have received the test done endpoint request, right?
NIT: maybe rename it to something like `removeListener`, `removeSubscription` etc. (otherwise it reads like it will start a subscription or something 😄 )